### PR TITLE
Feat: Implement web-provider package

### PR DESCRIPTION
**Pull request by teddyjfpender**
## Describe changes
To communicate with zkApps, Pallad must be able to receive requests and provide back data; Pallad must be able to talk! It must do so by following a specific pattern. A good implementation of the injected provider means developers spend little time integrating a wallet Pallad, and users of Pallad have granular control over what information they do and do not provide when their wallet receives a request.

This implementation is modelled around [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) with the `mina_` prefix instead of the `eth_` prefix.

Preliminary Mina-specific Tasks:
- [ ] add versioning of API
- [ ] `request` method 
    - [x] `mina_accounts` RPC Method
    - [x] `mina_sign` RPC Method
    - [x] `mina_signTransaction`  RPC Method
    - [x] `mina_signFields`  RPC Method
    - [x] `mina_signMessage` RPC Method (done in `mina_sign`)
    - [x] `mina_getBalance` RPC Method
    - [x]  `mina_chainId` RPC Method
    - [ ] `mina_sendTransaction` RPC Method
    - [ ] `mina_createNullifier` RPC Method
- [x] `enable` method
- [x] `connect` method
- [x] `disconnect` method
- [x] application listeners
    - [x] `on`
    - [x] `once`
    - [x] `removeListener`
    - [x] `off` 
- [ ] Correct event emitting
    - [ ] `accountsChanged`
    - [x] `chainChanged` 
- [x] Error handling 
    - [x] `4001`
    - [x] `4100`
    - [x] `4200`
    - [x] `4900`
    - [x]  `4901`
- [ ] Integrate into extension

Secondary Web3 Broad Tasks:
- [ ] Abstract over the `MinaProvider` to make a `UniversalProvider` for other ecosystems like Protoki, Zeko, Ethereum
- [ ] Add versioning

### CODE REVIEW

**Comment by teddyjfpender** [comment]
pls review

**Comment by teddyjfpender** [comment]
pls review

**Comment by teddyjfpender** [comment]
pls review

**Comment by teddyjfpender** [comment]
pls review

File: apps/extension/public/pallad_rpc.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 const BROADCAST_CHANNEL_ID = 'pallad'
 const callPalladAsync = ({ method, payload }) =>
   new Promise((resolve, reject) => {
+    console.log('callPalladAsync', method, payload)
```
Comment (mrcnk) [inline apps/extension/public/pallad_rpc.js:5]: remove logs pls

File: packages/web-provider/src/vault-service/vault-service.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    )
+  }
+
+  async sign(
+    signable: ChainSignablePayload,
+    args: ChainOperationArgs,
+    getPassphrase: GetPassphrase
+  ): Promise<unknown> {
+    const store = useVault.getState()
+    console.log('signable: ', signable)
```
Comment (mrcnk) [inline packages/web-provider/src/vault-service/vault-service.ts:40]: remove logs pls

File: packages/web-provider/src/universal-provider/types.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import { EventEmitter } from 'events'
```
Comment (mrcnk) [inline packages/web-provider/src/universal-provider/types.ts:1]: I'm not so sure we should add EventEmitter. It's nodejs native and we may get troubles in browser context. But if it works fine now, let it be.

File: packages/web-provider/package.json [hunk 4]
```diff
@@ -22,11 +22,14 @@
     "@palladxyz/mina-core": "^0.0.1",
     "@palladxyz/mina-graphql": "^0.0.1",
     "@palladxyz/multi-chain-core": "^0.0.1",
-    "@palladxyz/vault": "^0.0.1"
+    "@palladxyz/vault": "^0.0.1",
+    "mina-signer": "^2.1.1",
+    "vitest": "^1.1.0"
```
Comment (mrcnk) [inline packages/web-provider/package.json:27]: Vitest should be in `devDependencies`.

**Review by mrcnk — CHANGES_REQUESTED** [review]

**Comment by teddyjfpender** [comment]
TODO: Complete e2e scenarios.

**Pull request merged**